src/librustc_middle/ty/context.rs

```diff
@@ -1974,9 +1986,9 @@ impl<'tcx> Borrow<[Predicate<'tcx>]> for Interned<'tcx, List<Predicate<'tcx>>> {
     }
 }
 
-impl<'tcx> Borrow<Const<'tcx>> for Interned<'tcx, Const<'tcx>> {
-    fn borrow<'a>(&'a self) -> &'a Const<'tcx> {
-        &self.0
+impl<'tcx> Borrow<super::sty::InternedConst<'tcx>> for Interned<'tcx, Const<'tcx>> {
+    fn borrow<'a>(&'a self) -> &'a super::sty::InternedConst<'tcx> {
+        unsafe { mem::transmute(self) }
```

Comment:
`Const` and `InternedConst` may have a different layout.

Reply:
I thought it would be fine as they are both `repr(C)`.

Reply:
Missed the `#[repr(C)]`.